combat-trainer.lic

```diff
@@ -1287,6 +1358,30 @@ class SpellProcess
     end
   end
 
+  def check_trader_magic(game_state)
+    return unless DRStats.trader?
+    if Flags['ct-starlight-depleted'] then
+      echo "----OUT OF STARLIGHT - DELETING STARLIGHT MAGIC----"
+      @buff_spells.delete_if { |spell| @buff_spells[spell]['starlight'] }
+      @offensive_spells.delete_if do |spell|
+        if spell['starlight'] then true end
+      end
+      Flags['ct-cancel-regalia'] = true
+    end
+    if game_state.casting_regalia
+      if Flags['ct-starlight-depleted']  # In some conditions, regalia can run out of starlight partway through creation.
+        @regalia_array[game_state.swap_regalia_type].each do |item| 
+          bput("remove my #{item}", "into motes of silvery", "Remove what?", "You .*#{item}", 'The leather gauntlets slide')
+        end
+        @equipment_manager.wear_equipment_set?('standard')
+      end
+      Flags['ct-starlight-depleted'] ? game_state.last_regalia_type = nil : game_state.last_regalia_type = game_state.swap_regalia_type
```

Comment:
💡 Could this be `game_state.last_regalia_type = Flags['ct-starlight-depleted'] ? nil : game_state.swap_regalia_type`?

Reply:
Changed since that makes more sense.